### DISCUSSION TITLE
Build issues with current man branch, after milestone-7 release.

### DISCUSSION THREAD

**Discussion by FlorianSeidelAWS** [eclipse-edc/Connector#2271] [Q&A] [answered]
Hi all,

I am trying to build a connector launcher which has dependencies on the identity-hub, registration-service and federated-catalog projects. The built is identical to the connector launcher build in the MVD project, except that I am using the connector sub-projects from the main branch instead of the milestone-7 jars from maven. I get runtime errors when starting the launcher, which are all caused because the identity-hub, registration-service and federated-catalog projects all depend on connector sub-projects and pull in those dependencies from maven, which means that I have different versions of the same classes on the class path, which are not compatible. 
Questions:
1. How can I make this work? Are there any versions of the dependencies published which are compatible with the current main branch?
2. Is the code on the main branches currently compatible with each other? Could I build all the dataspace components locally and build a connector launcher from them that will work with the MVD?

**Comment by paullatzelsperger** [comment]
hi,
quite a few things changed between `0.0.1-milestone-7` and the current `0.0.1-SNAPSHOT` versions, so at this time they should not be expected to be interoperable, neither at compile-time, nor at runtime.

As of very recently, `federated-catalog`, `identity-hub` and `registration-service` already use the `0.0.1-SNAPSHOT` of the `connector`, but `mvd` still uses `milestone-7`.

Our plan is to upgrade MVD to also use the latest `SNAPSHOTs`, but we're still working on it. Aside from substantial code changes, this also entails a restructuring of our CI/CD pipelines so that we can guarantee compile-time and runtime compatibility of all components.

Eventually we want to be able to release all components with the same version, and have their `main` branches be compatible with each other, but for the time being, your best option is to go with the MVD dependency list, which is:
```
edcVersion=0.0.1-milestone-7
registrationServiceVersion=0.0.1-20221122-SNAPSHOT
identityHubVersion=0.0.1-20221118-SNAPSHOT
```

We hope to get this done until the [Milestone-8 release](https://github.com/eclipse-edc/Connector/milestone/10).

**Reply by FlorianSeidelAWS** [reply]
Hi Paul, thanks for the quick reply. I actually tried the version combination you proposed in your answer. It doesn't work. My conclusion is that it is not possible to build a connector launcher which has dependencies on the other components with the main connector branch. Our goal is to build an MVD with new AWS specific extensions. Our only option is to base them on the milestone-7 release of the connector project then and eventually port them to mileston-8, which will guarantee compile-time and runtime compatibility. Correct?

**Reply by paullatzelsperger** [reply]
That is what I meant, yes. You cannot use `connector - 0.0.1-SNAPSHOT` for MVD, you'll have to use the `milestone-7` release. 
What exactly is that AWS-Specific extension doing?

**Reply by FlorianSeidelAWS** [reply]
We plan to write extensions for SecretsManager, EventBridge and DynamoDB.

**Reply by paullatzelsperger** [reply]
ok, I don't know the intrinsic details of these, but I don't expect many breaking changes when porting that from `milestone-7` to `0.0.1-SNAPSHOT`. Our respective SPIs there have been fairly stable.

**Comment by paullatzelsperger** [comment, marked as answer]
FYI there is a [pull request](https://github.com/eclipse-edc/MinimumViableDataspace/pull/104) to upgrade MVD to use `0.0.1-SNAPSHOT` versions of all other components.
